src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -0,0 +1,335 @@
+        frontIntakeIR = map.frontIntakeIR();
+        bottomPierreIR = map.bottomPierreIR();
+        topPierreIR = map.topPierreIR();
+        backIntakeIR = map.backIntakeIR();
+
+        // bottomPierreIR = map.bottomPierreIR();
+        // topPierreIR = map.topPierreIR();
+        // backIntakeIR = map.backIntakeIR();
+        pierreMotor = map.pierreMotor();
+        
+    }
+
+    public SequentialCommandGroup intakeToPierre() {
+        return new SequentialCommandGroup(pierrePossesion(), runToClear());
+    }
+    
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered(){
+        final ParallelDeadlineGroup topIRStop = new ParallelDeadlineGroup(intakeToPierre(), pierrePossesion(),
+                singulatorPossesion(),
+                loadBallToTop(), runToClear());
+
+
+         topIRStop.setDeadline(stopWhenBallsAtTop());
+
+         return topIRStop;
+    }
+    //This will stop all commands when the top IR sensor on Pierre is triggered
+
+    public CommandBase indexMotor(final double motorSpeed) {
+        return new StartEndCommand(() -> {
+            singulator.set(motorSpeed);
+        }, () -> {
+            singulator.set(0);
+        });
+    }
+
+    public CommandBase runPierre() {
+        return new StartEndCommand(() -> {
+            pierreMotor.set(.75);
+        }, () -> {
+            pierreMotor.stopMotor();
+        }, this);
+    }
+
+
+
+    public CommandBase quicklyPush() {
+        return indexMotor(singulatorMotorSpeed);
+    }
+
+    public CommandBase reversePush() {
+        return indexMotor(-singulatorMotorSpeed);
+    }
+
+    public CommandBase quicklyOutput() {
+        return indexMotor(pierreIndexSpeed);
+    }
+
+    public CommandBase reverseOutput() {
+        return indexMotor(-pierreIndexSpeed);
+    }
+
+    /*
+     * Command to run the singlator motor until the IR sensor 1 (analog 0) Command
+     * to run the singulator and pierre motor until the IR senor 1 is empty and IR
+     * senor 2 is covered Command to run the pierre motor until the IR sensor is
+     * uncovered
+     */
+    public CommandBase singulatorPossesion() {
+        return new CommandBase() {
+            {
+                addRequirements(Indexer.this);
+            }
+
+
+            @Override
+            public boolean isFinished() {
+                return frontIntakeIR.getAsBoolean();
+                
+                //frontIntakeIR.getVoltage() > irSensorVoltage;
+                // values of .8 when open, 1.7 when closed
+
+            }
+
+            @Override
+            public void execute() {
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                singulator.set(0);
+            }
+
+        };
+
+    }
+    //This command will make sure that the singulator has possesion of the ball
```

Comment:
Comments should go before the code they reference or on the same line, not afterwards.